salt/orch/removal.sls

```diff
@@ -204,9 +206,11 @@ sync-all:
       - saltutil.sync_all
     - require:
       - assign-removal-grain
+{%- if replacement %}
   {%- for role in replacement_roles %}
       - assign-{{ role }}-role-to-replacement
   {%- endfor %}
+{%- endif %}
```

Comment:
So was this failing beacuse there was no `replacement` but there were `replacement_roles`?

Reply:
Yes, exactly.

Reply:
In that case it doesn't make sense to have some roles for a replacement that does not exist. Maybe it would be better to fix it [here](https://github.com/kubic-project/salt/blob/master/salt/orch/removal.sls#L152-L157), with something like:

```Jinja
{%- if replacement %}
  {#- consistency check: if there is a replacement, it must have some (new) role(s) #}
  {%- if not replacement_roles %}
    {%- do salt.caasp_log.abort('CaaS: %s cannot be removed: too few etcd members, and no replacement found', target) %}
  {%- endif %}
{%- else %}
    {#- remove the roles for a replacement that could not be found #}
  {%- set replacement_roles = [] %}
{%- endif %} {# replacement #}
```

Reply:
What you propose would make sense, let me check the original problem again. Will update the PR.

Reply:
When removing a master, the commit https://github.com/kubic-project/salt/pull/433/commits/0196d659fc1e5565706f53a37022149e055a68ff is fixing this error:

```
caasp-admin_master:
    Data failed to compile:
----------
    Rendering SLS 'base:orch.removal' failed: Jinja variable 'replacement_provided' is undefined
```